src/scopes/trainingReports/goalName.ts

```diff
@@ -0,0 +1,25 @@
+import { Op } from 'sequelize';
+
+export function withGoalName(searchText: string[]) {
+  const [search] = [`%${searchText.map((st) => st.toLowerCase())}%`];
+
+  return {
+    [Op.and]: {
+      'data.goal': {
+        [Op.iLike]: search,
```

Comment:
This is good prevention for sql injection, thank you. Do we need to make other stories to prevent sql injection from our other filters?

Reply:
I think we can move this discussion to the #2029, but short answer is I don't think we have any obvious SQL injection vulnerabilities right now. If you validate that there is one, yeah, we should write a story.